test/tests/serial/mail_hostname.bats

```diff
@@ -1,213 +1,260 @@
 }
 
-@test "checking SRS: OVERRIDE_HOSTNAME is handled correctly" {
-  run docker exec mail_override_hostname grep "SRS_DOMAIN=my-domain.com" /etc/default/postsrsd
-  assert_success
-}
+function _should_have_expected_hostname() {
+  local EXPECTED_FQDN=${1}
 
-@test "checking dovecot: postmaster address" {
-  run docker exec mail_override_hostname /bin/sh -c "grep 'postmaster_address = postmaster@my-domain.com' /etc/dovecot/conf.d/15-lda.conf"
+  _run_in_container_bash "hostname"
+  assert_output "${EXPECTED_FQDN}"
   assert_success
-}
 
-#
-# non-subdomain tests
-#
-
-@test "checking configuration: non-subdomain: check container hostname is applied correctly" {
-  run docker exec mail_non_subdomain_hostname /bin/bash -c "hostname | grep domain.com"
+  _run_in_container_bash "grep -E '[[:space:]]+${EXPECTED_FQDN}' /etc/hosts"
   assert_success
 }
 
-@test "checking configuration: non-subdomain: check overriden hostname is applied to all configs" {
-  run docker exec mail_non_subdomain_hostname /bin/bash -c "cat /etc/mailname | grep domain.com"
-  assert_success
+function _should_be_configured_to_domainname() {
+  local EXPECTED_DOMAIN=${1}
 
-  run docker exec mail_non_subdomain_hostname /bin/bash -c "postconf -n | grep mydomain | grep domain.com"
+  # setup-stack.sh:_setup_mailname
+  _run_in_container_bash "cat /etc/mailname"
+  assert_output "${EXPECTED_DOMAIN}"
   assert_success
 
-  run docker exec mail_non_subdomain_hostname /bin/bash -c "postconf -n | grep myhostname | grep domain.com"
+  # Postfix
+  _run_in_container_bash "postconf mydomain"
+  assert_output "mydomain = ${EXPECTED_DOMAIN}"
   assert_success
 
-  run docker exec mail_non_subdomain_hostname /bin/bash -c "doveconf | grep hostname | grep domain.com"
+  # PostSRSd
+  _run_in_container_bash "grep '^SRS_DOMAIN=' /etc/default/postsrsd"
+  assert_output "SRS_DOMAIN=${EXPECTED_DOMAIN}"
   assert_success
 
-  run docker exec mail_non_subdomain_hostname /bin/bash -c "cat /etc/opendmarc.conf | grep AuthservID | grep domain.com"
+  # Dovecot
+  _run_in_container_bash "grep '^postmaster_address' /etc/dovecot/conf.d/15-lda.conf"
+  assert_output "postmaster_address = postmaster@${EXPECTED_DOMAIN}"
   assert_success
+}
 
-  run docker exec mail_non_subdomain_hostname /bin/bash -c "cat /etc/opendmarc.conf | grep TrustedAuthservIDs | grep domain.com"
-  assert_success
+function _should_be_configured_to_fqdn() {
+  local EXPECTED_FQDN=${1}
 
-  run docker exec mail_non_subdomain_hostname /bin/bash -c "cat /etc/amavis/conf.d/05-node_id | grep myhostname | grep domain.com"
+  # Postfix
+  _run_in_container_bash "postconf myhostname"
+  assert_output "myhostname = ${EXPECTED_FQDN}"
   assert_success
-}
-
-@test "checking configuration: non-subdomain: check hostname in postfix HELO message" {
-  run docker exec mail_non_subdomain_hostname /bin/bash -c "nc -w 1 0.0.0.0 25 | grep domain.com"
+  # Postfix HELO message should contain FQDN (hostname)
+  _run_in_container_bash "nc -w 1 0.0.0.0 25"
```

Comment:
Really not sure about this one...

Reply:
If you meant converting to `_run_in_container nc -w 1 0.0.0.0 25` it passes the assertion :+1: